packages/web-app-files/src/components/SideBar/Shares/Collaborators/RoleDropdown.vue

```diff
@@ -19,6 +19,11 @@
         <span class="oc-text-truncate" v-text="inviteLabel" />
         <oc-icon name="arrow-down-s" />
       </oc-button>
+      <oc-contextual-helper
+        v-if="isDisabledRole"
+        class="oc-ml-xs files-permission-actions-list"
+        :list="existingSharePermissions.map((permission) => ({ text: $gettext(permission) }))"
```

Comment:
We still need add the list of permissions to the `additionalTranslations.ts` file, otherwise they don't land in Transifex. You can use the values from the `GraphSharePermission` enum as keys.

Reply:
thanks for the heads up, done, but looks a bit loose to me... something that is easy to forget, isnt it?

i thing something like

```js
// pseudo code
existingSharePermissions.map((permission) => ({ text: additionalTranslations.knownSharePermissions[permission] }))
```

or similar won't work as expected? right?

Reply:
> thanks for the heads up, done, but looks a bit loose to me... something that is easy to forget, isnt it?

There are only a few cases, but it can be overseen quite easily, yeah :/

> i thing something like
> 
> ```js
> // pseudo code
> existingSharePermissions.map((permission) => ({ text: additionalTranslations.knownSharePermissions[permission] ?? $gettext(permission) }))
> ```
> 
> or similar won't work as expected? right?

Correct, this again wouldn't land in Transifex. The extractor only looks for strings inside of `$gettext()`.